include/votca/csg/interaction.h

```diff
@@ -100,224 +118,62 @@ class Interaction {
   void RebuildName();
 };
 
-inline void Interaction::RebuildName() {
-  std::stringstream s;
-  if (_mol != -1) {
-    { s << "molecule " << _mol; }
-  }
-  if (!_group.empty()) {
-    s << ":" << _group;
-    if (_group_id != -1) {
-      s << " " << _group_id;
-    }
-  }
-  if (_index != -1) {
-    { s << ":index " << _index; }
-  }
-  _name = s.str();
-}
-
 /**
     \brief bond interaction
 */
 class IBond : public Interaction {
  public:
-  IBond(Index bead1, Index bead2) {
-    _beads.resize(2);
-    _beads[0] = bead1;
-    _beads[1] = bead2;
-  }
-
-  IBond(std::list<Index> &beads) {
-    assert(beads.size() >= 2);
-    _beads.resize(2);
-    for (Index i = 0; i < 2; ++i) {
-      _beads[i] = beads.front();
-      beads.pop_front();
-    }
-  }
-  double EvaluateVar(const Topology &top) override;
-  Eigen::Vector3d Grad(const Topology &top, Index bead) override;
+  virtual ~IBond() = default;
+  double EvaluateVar(const Topology &top) const override;
+  Eigen::Vector3d Grad(const Topology &top, Index bead) const override;
 
  private:
+  template <class T>
```

Comment:
Why hiding the contructor?

Reply:
This couples the interaction to topology.

Reply:
Yes, and makes it more consistent with the design. I already responded in slack, so we can go from there.